data/spec/SBOM.md

```diff
@@ -1,3 +1,312 @@
-# Data spec for SBOM
+# SBOM Spec
 
-TBD
\ No newline at end of file
+The specification between harbor and scanners, used to define a unified standard for SBOM interaction between Harbor and third-party scanners, for vendor integration and compatibility.
+
+## MIME Types
+
+- `application/vnd.security.sbom.report+json; version=1.0`
+- `application/vnd.goharbor.harbor.sbom.v1+json`
+
+## Statement
+
+Since this specification is built upon the original scanner specification, it is recommended to implement the SBOM specification on the basis of implementing the original v1.1 scanner specification, but the SBOM specification is an **OPTIONAL** specification, which only expands the capabilities of the original specification. For manufacturers without SBOM requirements, implementation may not be required.
+
+### Capabilities
+
+#### Generate SBOM
+
+The capability to generate the SBOM from the container image, the process is similar with the scan vulnerabilities, the scanner should pull the image from harbor and then generate SBOM, harbor will polling the scanner to get the SBOM until timeout or error occurred. Supported produced MIME types should includes the `application/vnd.security.sbom.report+json; version=1.0`.
+
+#### Scan SBOM
+
+The capability to scan the image vulnerabilities from the SBOM of image, which have the better performance as the scanner only needs to pull the SBOM artifact instead of whole image. If the `config.mediaType` is `application/vnd.goharbor.harbor.sbom.v1+json` of the manifest, then scanner should treat is as SBOM artifact and pull the raw SBOM content from layers and scan vulnerabilities from it, the `artifact` field in the report should be the **subject** artifact.
```

Comment:
Does it mean users select which SBOM they want to scan in Harbor UI? According to OCI 1.1, multiple SBOM reports can refer to the same container image. 

The scanner doesn't need to select SBOM, but the user does it, and Harbor sends the digest to the scanner, right?

Reply:
Yes, the user may select or harbor will choose a default one then sends the digest to the scanner, the scanner does not need to care about the number of the SBOMs attached to the image, the scanner only needs to pull the first layer defined in the layers and scan it as SBOM when the `manifest.artifactType` is `application/vnd.goharbor.harbor.sbom.v1`.

Reply:
Doesn't Harbor tell scanners `media_type`? Does the scanner need to identify the format?

Reply:
Anyway, the scanner needs to pull the artifact manifest firstly so the `artifactType` can be found in the manifest if it is a SBOM artifact, harbor also can retrieve it and field it into the scan request but that may be redundant.

Reply:
OK, we will see `artifactType` in the manifest, then. Thanks!